documentation/edit_installation/post-installation

wabernat opened the pull request:
Deep edit and process walk-through of post-install steps.

Updated conf.py to show year 2020

Updated links and reference targets in other docs.

**Component**:

documentation

**Context**: 

**Summary**:

I walked the procedure as a naive operator. Made process and documentation changes based on results.

**Acceptance criteria**: Looks good? Is good.

---

See: #ZENKO-2421

wabernat committed:
Walked the steps on the bootstrap process. Made changes to conform to my naive
user eperience. Simplified some of the commands. Gave (I hope) correct
orthography on root (#) vs. user ($) command prompts.

Reordered chapter (index.rst) to make the procedure make more sense: Users
cannot access GUI until PVs are allocated for Prometheus and Alertmanager, so
PV activation must come *before* GUI instructions.

Trivial change to introduction (for euphony).

wabernat committed:
deep edit and process walk-through of post-install steps.

updated conf.py to show year 2020

updated links and reference targets in other docs.

bert-e commented:
# Hello wabernatscality,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/2.5/documentation/post-install_edit` with contents from `documentation/post-install_edit`
and `development/2.5`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/2.5/documentation/post-install_edit origin/development/2.5
 $ git merge origin/documentation/post-install_edit
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.5/documentation/post-install_edit
```

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/2.6/documentation/post-install_edit` with contents from `w/2.5/documentation/post-install_edit`
and `development/2.6`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/2.6/documentation/post-install_edit origin/development/2.6
 $ git merge origin/w/2.5/documentation/post-install_edit
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.6/documentation/post-install_edit
```

slaperche-scality commented on docs/conf.py:
```diff
@@ -24,7 +24,7 @@
 
 project = 'MetalK8s'
-copyright = '2019, Scality'
+copyright = '2020, Scality'
```
To avoid an update every year, I would go with:

```python
copyright = '{}, Scality'.format(dt.datetime.now().year)
```

This will require a `import datetime as dt` to be added near the other import statements.

wabernat replied:
I want to address this in a separate ticket.

NicolasT commented on docs/developer/architecture/ci.rst:
```diff
@@ -0,0 +1,168 @@
+Continuous Testing
```
Not sure how this became part of this PR? Shouldn't be in it in any case.

wabernat replied:
yeah.... I have no recall of this. I suspect my Git "skills" may have contributed. I did have a delay and may have pulled this in on a merge. 

Rebased to development/2.4. This looks to have gone away.

NicolasT commented on docs/installation/post-install.rst:
```diff
@@ -1,23 +1,23 @@
-Post-Installation Procedure
-===========================
+After Installation
```
May make sense to then also rename the file.

NicolasT commented on docs/installation/post-install.rst:
```diff
@@ -1,23 +1,23 @@
+<node-role-bootstrap>` from :ref:`Infra services <node-role-infra>` is
+recommended.
+
+To create the required "Volume" objects, copy the following code to a YAML
```
We should have some semantic role for object types... Just putting them between quotes doesn't feel right.

NicolasT commented on docs/installation/post-install.rst:
```diff
@@ -26,12 +26,12 @@ path for the partitions to use:
    kind: Volume
    metadata:
-     name: <node_name>-prometheus
+     name: bootstrap-prometheus
```
No, this would only be the case when deploying Prometheus and AlertManager on the bootstrap node, which we explicitly recommend against above.

NicolasT commented on docs/installation/post-install.rst:
```diff
@@ -26,12 +26,12 @@ path for the partitions to use:
      storageClassName: metalk8s-prometheus
-     rawBlockDevice:  # Choose a device with at least 10GiB capacity
-       devicePath: <device_path>
+     sparseLoopDevice:
```
No, not at all. While yes this may *work*, it's definitely *not* what we want people to do in production setups.

**We can not give copy-paste instructions**. People have to *understand* what they are doing and *make decisions*, adapting certain things (like these templates) to any such decisions they made.

NicolasT commented on docs/installation/post-install.rst:
```diff
@@ -40,63 +40,81 @@ path for the partitions to use:
    kind: Volume
    metadata:
-     name: <node_name>-alertmanager
+     name: bootstrap-alertmanager
```
Same

NicolasT commented on docs/installation/post-install.rst:
```diff
@@ -40,63 +40,81 @@ path for the partitions to use:
      storageClassName: metalk8s-prometheus
-     rawBlockDevice:  # Choose a device with at least 1GiB capacity
-       devicePath: <device_path2>
+     sparseLoopDevice:
```
Same

NicolasT commented on docs/installation/post-install.rst:
```diff
@@ -40,63 +40,81 @@ path for the partitions to use:
 
-   root@bootstrap $ kubectl --kubeconfig /etc/kubernetes/admin.conf \
-                      apply -f <file_path>
+   root@bootstrap # kubectl --kubeconfig /etc/kubernetes/admin.conf apply -f <file_path>
```
This was also 'wrong' in the original: we should at all times assume `KUBECONFIG` is exported and not pass `--kubeconfig ...` every time to reduce duplication / useless overhead.

NicolasT commented on docs/installation/post-install.rst:
```diff
@@ -40,63 +40,81 @@ path for the partitions to use:
-:doc:`this section of the Operational Guide
-<../operation/volume_management/index>`.
+:doc:`Volume Management <../operation/volume_management/index>` in
+:ref:`MetalK8s Operation <../operation/index>`.
```
Do we really need the `in MetalK8s Operation`? What's wrong with `see Volume Management.`?

NicolasT commented on docs/installation/post-install.rst:
```diff
@@ -40,63 +40,81 @@ path for the partitions to use:
+GUI Validation
+^^^^^^^^^^^^^^
+
+If things are installed correctly, you can open and log in to the new bootstrap
```
You're not logging in to the bootstrap server.

NicolasT commented on docs/installation/post-install.rst:
```diff
@@ -40,63 +40,81 @@ path for the partitions to use:
+If things are installed correctly, you can open and log in to the new bootstrap
+server.
+
+Open your browser and access the bootstrap server. Enter the bootstrap server's
```
Not the bootstrap server. You're accessing the MetalK8s UI.

NicolasT commented on docs/installation/post-install.rst:
```diff
@@ -106,21 +124,20 @@ applications, perform the following sanity checks:
-   ``--all-namespaces`` flag. On the other hand, use the ``-n`` or
-   ``--namespace`` option to select Pods in a given :term:`Namespace`.
+   To look for all Pods at once, use the ``--all-namespaces`` flag. To select
+   Pods in a given :term:`Namespace` use the ``-n`` or ``--namespace`` option
```
Missing `.`

NicolasT commented on docs/installation/post-install.rst:
```diff
@@ -150,20 +167,21 @@ applications, perform the following sanity checks:
    .. code-block:: shell
 
-      root@bootstrap $ kubectl --kubeconfig /etc/kubernetes/admin.conf \
+      root@bootstrap # kubectl --kubeconfig /etc/kubernetes/admin.conf\
```
Why remove the space?

wabernat committed:
Merge branch 'development/2.4' of http://github.com/Scality/metalk8s into development/2.4

wabernat committed:
documentation: clarified bootstrap instructions further.

bert-e commented:
# History mismatch



Merge commit #6ebb08892e94e8f3a018aae8293709e73e984fff on the integration branch
`w/2.5/documentation/post-install_edit` is merging a branch which is neither the current
branch `documentation/post-install_edit` nor the development branch
`development/2.5`.

It is likely due to a rebase of the branch `documentation/post-install_edit` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

wabernat committed:
documentation/edit_installation/post-installation

Rebased to correct branch.

Implemented all requested changes.

Changed title lines to consistent (imperative) mood.

wabernat commented:
@bert-e reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/2.5/documentation/post-install_edit` with contents from `documentation/post-install_edit`
and `development/2.5`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/2.5/documentation/post-install_edit origin/development/2.5
 $ git merge origin/documentation/post-install_edit
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.5/documentation/post-install_edit
```

wabernat committed:
Fixed linter errors

wabernat reviewed (commented):
implemented or deferred requested changes.

bert-e commented:
# Branches have diverged



This pull request's source branch `documentation/post-install_edit` has diverged from
`development/2.4` by more than 50 commits.

To avoid any integration risks, please re-synchronize them using one of the
following solutions:

* Merge `origin/development/2.4` into `documentation/post-install_edit`
* Rebase `documentation/post-install_edit` onto `origin/development/2.4`

Note: If you choose to rebase, you may have to ask me to rebuild
integration branches using the `reset` command.

wabernat committed:
Merge branch 'development/2.4' of http://github.com/Scality/metalk8s into development/2.4

wabernat committed:
Fixed a stealthy git conflict.

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/2.5/documentation/post-install_edit` with contents from `documentation/post-install_edit`
and `development/2.5`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/2.5/documentation/post-install_edit origin/development/2.5
 $ git merge origin/documentation/post-install_edit
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.5/documentation/post-install_edit
```

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/2.6/documentation/post-install_edit` with contents from `w/2.5/documentation/post-install_edit`
and `development/2.6`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/2.6/documentation/post-install_edit origin/development/2.6
 $ git merge origin/w/2.5/documentation/post-install_edit
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.6/documentation/post-install_edit
```

wabernat commented:
@NicolasT I have implemented requested changes, but this PR looks to be hung up on a conflict that I cannot resolve by following the instructions I'm being offered. 

Any sage counsel for pushing this thing through to get an integration branch?

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @nootal

* @alexandre-allard-scality

* @lucieleonard

* @sayf-eddine-scality

* @slaperche-scality

* @nicolast

* @gdemonet

* @carlito-scality

* @thomasdanan

* @jbertran

* @teddyandrieux

* @cmonfort

* @monpote





The following reviewers are expecting changes from the author, or must review again:

* @nicolast

wabernat commented:
Ok... that was weird, but I'll take it.

wabernat committed:
Merge branch 'development/2.4' into documentation/post-install_edit

bert-e commented:
# Conflict




A conflict has been raised during the update of
integration branch `w/2.5/documentation/post-install_edit` with contents from `documentation/post-install_edit`
and `development/2.5`.

Please resolve the conflict on **the integration branch** (`w/2.5/documentation/post-install_edit`).


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout w/2.5/documentation/post-install_edit
 $ git pull  # or "git reset --hard origin/w/2.5/documentation/post-install_edit"
 $ git merge origin/development/2.5
 $ # <intense conflict resolution>
 $ git commit
 $ git merge origin/documentation/post-install_edit
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.5/documentation/post-install_edit
```

bert-e commented:
# Conflict




A conflict has been raised during the update of
integration branch `w/2.6/documentation/post-install_edit` with contents from `w/2.5/documentation/post-install_edit`
and `development/2.6`.

Please resolve the conflict on **the integration branch** (`w/2.6/documentation/post-install_edit`).


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout w/2.6/documentation/post-install_edit
 $ git pull  # or "git reset --hard origin/w/2.6/documentation/post-install_edit"
 $ git merge origin/development/2.6
 $ # <intense conflict resolution>
 $ git commit
 $ git merge origin/w/2.5/documentation/post-install_edit
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.6/documentation/post-install_edit
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @nootal

* @alexandre-allard-scality

* @lucieleonard

* @sayf-eddine-scality

* @slaperche-scality

* @nicolast

* @gdemonet

* @carlito-scality

* @thomasdanan

* @jbertran

* @teddyandrieux

* @cmonfort

* @monpote





The following reviewers are expecting changes from the author, or must review again:

* @nicolast

wabernat commented:
@bert-e approve

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @nootal

* @alexandre-allard-scality

* @lucieleonard

* @sayf-eddine-scality

* @slaperche-scality

* @nicolast

* @gdemonet

* @carlito-scality

* @thomasdanan

* @jbertran

* @teddyandrieux

* @cmonfort

* @monpote





The following reviewers are expecting changes from the author, or must review again:

* @nicolast







*The following options are set:* **approve**
